mitodl/open-discussions
# Improvements to moderator/contributor UI

#### What are the relevant tickets?
Fixes #1004 
Fixes #971 
Fixes #938 

#### What's this PR do?
 - Adds "Can't remove" text if the moderator is not removable due to being older than logged in user
 - Adds confirmation dialog for removing a user
 - Changes in text when the user you're removing is yourself

#### How should this be manually tested?
Go to the moderators page. You should likely see some moderator which says "Can't remove" instead of a "Remove" link. If not, add a new moderator, log in as that moderator and view the list again.

You should also see `You are a moderator of this channel` in the list, and `Leave` instead of `Remove` in the text. Click the link and you should see a dialog "Remove yourself?"

Find a user you can remove and remove them via the dialog to confirm that behavior still works. You should also be able to dismiss the dialog without anyone getting removed.

All of this should be exactly the same for contributors
